mixer/main.go

```diff
@@ -1,270 +1,21 @@
-package main
-
-import (
-	"flag"
-	"fmt"
-	"io/ioutil"
-	"os"
-	"os/exec"
-	"strconv"
-	"strings"
-
-	"github.com/clearlinux/mixer-tools/builder"
-	"github.com/clearlinux/mixer-tools/helpers"
-)
-
-type Command struct {
-	Name        string
-	Description string
-	Run         func(args []string)
-}
-
-var commands []*Command
-
-func init() {
-	commands = []*Command{
-		{"build-all", "Build all content for mix with default options", cmdBuildAll},
-		{"build-chroots", "Build chroots for the mix", cmdBuildChroots},
-		{"build-update", "Build all update content for the mix", cmdBuildUpdate},
-		{"build-image", "Build an image from the mix content", cmdBuildImage},
-		{"add-rpms", "Add rpms to local yum repository", cmdAddRPMs},
-		{"get-bundles", "Get the clr-bundles from upstream", cmdGetBundles},
-		{"add-bundles", "Add clr-bundles to your mix", cmdAddBundles},
-		{"init-mix", "Initialize the mixer and workspace", cmdInitMix},
-		{"help", "Show help options", cmdHelp},
-	}
-}
+// Copyright © 2017 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
-func PrintMainHelp() {
-	fmt.Printf("usage: mixer <command> [args]\n")
-	for _, cmd := range commands {
-		fmt.Printf("\t%-20s\t%s\n", cmd.Name, cmd.Description)
-	}
-}
+package main
 
-func CheckDeps() error {
-	deps := []string{
-		"createrepo_c",
-		"git",
-		"hardlink",
-		"m4",
-		"openssl",
-		"parallel",
-		"rpm",
-		"yum",
-	}
-	for _, dep := range deps {
-		if _, err := exec.LookPath(dep); err != nil {
-			return fmt.Errorf("failed to find program %q: %v\n", dep, err)
-		}
-	}
-	return nil
-}
+import "github.com/clearlinux/mixer-tools/mixer/cmd"
 
 func main() {
-	fmt.Println("Mixer 3.2.0")
```

Comment:
How can I query the version of mixer now?

Reply:
Mistake.

Reply:
Cobra will auto-generate a --version flag if you set the `Version` field on the root command